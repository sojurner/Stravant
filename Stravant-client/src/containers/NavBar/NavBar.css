.nav-bar {
  display: flex;
  justify-content: space-between;
  font-size: 2.5rem;
  margin: 1rem auto; }

.nav-link {
  margin: 0 0.8rem;
  font-family: daybold;
  text-decoration: none;
  border: 1.5px solid orange;
  border-radius: 8px;
  padding: 0.5rem 0.5rem 0 0.5rem;
  border-style: inset; }

.nav-link:hover {
  background-color: orange; }

.active {
  background-color: #b8995f; }
