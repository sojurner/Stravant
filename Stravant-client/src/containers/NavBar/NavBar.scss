.nav-bar {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  font-size: 2.5rem;
  margin: 1rem auto;
  background-color: #fc8e6b;
  width: 100vw;
  margin: 0;
  height: 5.5rem;
}

.nav-link {
  margin: 1.5rem 0.8rem;
  font-family: daybold;
  text-decoration: none;
  padding: 0.2rem 0.5rem 0.2rem 0.5rem;
}

.nav-link:hover {
  border-bottom: 3px solid #882709;
}

.active {
  background-color: rgb(248, 227, 188);
  border-radius: 5px;
}

@media screen and (max-width: 1050px) {
  .nav-link {
    font-size: 4vw;
  }
}

@media screen and (max-width: 880px) {
  .nav-link {
    font-size: 2rem;
  }
}

@media screen and (max-width: 760px) {
  .nav-link {
    font-size: 3vw;
    padding-top: 0.6rem;
  }
}

@media screen and (max-width: 510px) {
  .nav-link {
    font-size: 0.8rem;
  }
}

// @media screen and;
