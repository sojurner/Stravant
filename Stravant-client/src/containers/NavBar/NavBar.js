import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import './NavBar.css';

export const NavBar = () => {
  const navLinks = ['personal', 'weekly', 'leaderboard', 'pomodoro'];
  const displayLinks = navLinks.map((link, index) => {
    const navClass = `nav-link nav-${link}`;
    return (
      <NavLink key={index} exact to={`/${link}`} className={navClass}>
        {link}
      </NavLink>
    );
  });
  return <nav className="nav-bar">{displayLinks}</nav>;
};
