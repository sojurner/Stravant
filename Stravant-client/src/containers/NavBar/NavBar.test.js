import React from 'react';
import { NavBar } from './NavBar';

describe('NavBar', () => {
  let wrapper = shallow(<NavBar />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
