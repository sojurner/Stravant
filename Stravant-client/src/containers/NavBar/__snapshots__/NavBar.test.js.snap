// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match snapshot 1`] = `
<nav
  className="nav-bar"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav-link nav-personal"
    exact={true}
    to="/personal"
  >
    personal
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav-link nav-weekly"
    exact={true}
    to="/weekly"
  >
    weekly
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav-link nav-leaderboard"
    exact={true}
    to="/leaderboard"
  >
    leaderboard
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav-link nav-pomodoro"
    exact={true}
    to="/pomodoro"
  >
    pomodoro
  </NavLink>
</nav>
`;
